### PR TITLE
Added tolerations, nodeSelector and nodeAffinity to the deployment of SMI adaptor

### LINKED ISSUE #10933: Add support for nodeselectors and tolerations in linkerd-smi helm chart
### What problem are you trying to solve?

I need to make sure that linkerd-smi runs in a specific pool where we only have critical system services running.

### How should the problem be solved?

Add nodeSelectors and tolerations to the Helm chart values file

### Any alternatives you've considered?

Decompile the helm charts into yaml, and use Kustomize to patch it

### How would users interact with this feature?

_No response_

### Would you like to work on this feature?

no

### PR DESCRIPTION
**Pull request by jclegras**
Custom values I used for validation:
```
---
adaptor:
  nodeSelector:
    workload: "critical"
  tolerations:
   - effect: "NoSchedule"
     key: "dedicated"
     operator: "Equal"
     value: "critical"
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchExpressions:
          - key: "workload"
            operator: In
            values:
            - "critical"
```

Fixes [#10933](https://github.com/linkerd/linkerd2/issues/10933)

### CODE REVIEW

File: charts/linkerd-smi/templates/adaptor.yaml [hunk 1]
```diff
@@ -21,6 +21,19 @@ spec:
...
+{{toYaml . | indent 8}}
+      {{- end }}
+      {{- with .Values.adaptor.nodeSelector }}
+      nodeSelector: 
+{{toYaml . | indent 8}}
+      {{- end }}
+      {{- with .Values.adaptor.affinity.nodeAffinity }}
+      affinity:
+        nodeAffinity:
+{{toYaml . | indent 10 }}
```
Comment (alpeb) [inline charts/linkerd-smi/templates/adaptor.yaml]: After having some experience with the affinity fields in the linkerd2 repo, I think the clearest is to have a single free-form `values.adaptor.affinity` setting, perhaps with usage examples in the entry doc in the values.yaml file. That way people can either declare `nodeAffinity` and `podAntiAffinity` at will.

**Review by alpeb — COMMENTED** [review]
Thanks @jclegras :+1: 
Can you please take care of the linter errors?

File: charts/linkerd-smi/templates/adaptor.yaml (hunk 1 above)
Comment (jclegras) [reply charts/linkerd-smi/templates/adaptor.yaml]: Fine for me too @alpeb, thanks for the feedback

**Review by alpeb — APPROVED** [review]
Thanks @jclegras, this looks good to me :+1:

**Pull request merged**